### DISCUSSION TITLE
What would make it a version 1.0?

### DISCUSSION THREAD

**Discussion by MothNik** [paucablop/chemotools#127] [General]
I've seen a lot of issues being tagged with the __version 0.2__ - milestone, but I think we should look at the bigger picture here:

**What would be required to make `chemotools` a version 1.0?**

Knowing what we want to reach in the far future will help us settle the roadmap for the tiny steps in between, i.e., knowing what version 1.0 should do could maybe already be prepared in version 0.2 🤯 

So, from my view, I think that `chemotools` is already on a very good track (way better than what a version 0.1 might suggest actually) and the only things for version 1.0 that immediately come into my mind would be:
- consistent typing, input validation, and module structure
- having all automated tests run in a CI pipeline
- "deprecating" Savitzky-Golay or at least making it good practice to use better alternatives
- addition of a deconvolution module
- automated hyperparameter selection whenever possible, especially for smoothing and baseline corrections as well (erPLS)
- making the setup more consistent and also enable faster numerics by better support for Numba and Cython and maybe beyond

What do you think? I'm really excited to hear about 💡 ideas on how to make `chemotools` the standard library for chemometric tasks 💪 🤓

**Comment by paucablop** [comment]
Hi @MothNik completely agree with you, when I planned version 0.2.0, nearly a year ago, I wanted to incorporate a bit more feedback from the users, to make sure that the package was ready, but I think that now we are passed that point and we should definitely aim at a version 1.0, maybe as a summer project! :palm_tree: :surfer: 

We could work on drafting a new milestone to develop the version 1.0. 

- I completely agree on reviewing the implementation to ensure consistent typing, validation of the input, and also rethinking the structure of the module. Also, aligning with some of the new test frameworks. 
- The are automated test on a CI pipeline but are only fired from main, we should extend these to include also be enabled when pushed from branches. Also, I would like to configure the CI pipeline before a pull request. :smile: 
- Great idea incorporating improved methods to the traditional filters!
- YEEAH i love the idea of including the deconvolution module, pleaase! super nice!! also have been considering implementing different flavors of MCR (what do you think about this).
- Super with automated hyperparameter selection!! 
- It would be very nice to keep investigating newer methods to enable faster numerics, I am very interested in that. By the way, do you have any experience using [jax](https://jax.readthedocs.io/en/latest/quickstart.html), I would love to hear your opinions about it!

Some additional thoughts for a version 1.0
- I would like to release the package to conda too, many users have requested that. 
- Several users have also requested to incorporate traditional residual analysis, this is something that could be useful for several chemometricians!! :)
- Also, I would like to incorporate some variable selection methods maybe starting by continuing with #58 as starting point!
- There are also some methods that need a bit more love and fix minor things, for instance: RangeCut requires an ordered ascending wavenumbers array, I would like to raise an exception when the wavenumber arrays are in descending order. 
- I have also been considering in adding an auto alignment module, to include different auto alignment algorithms (what do you think?)

Maybe we could see what can be prioritized for a version 1.0 and then start writing the Milestone and organizing some of the issues :)
